test/unittest/main.cpp

```diff
@@ -25,11 +25,46 @@
 
 #include <gtest/gtest.h>
 
+#include "blas_test.hpp"
 #include "blas_test_macros.hpp"
+#include "clara.hpp"
 
-int main(int argc, char *argv[]) {
+struct Args args{};
+
+namespace {
+/**
+ * @fn parse_args
+ * @brief Returns a structure containing the data extracted from the
+ * command-line arguments.
+ */
+void parse_args(int argc, char** argv) {
+  // Updates the global args
+  args.program_name = std::string(argv[0]);
+  bool show_help;
+
+  auto parser = clara::Help(show_help) |
+                clara::Opt(args.device, "device")["--device"](
+                    "Select a device (best effort) for running the benchmark.");
+
+  auto res = parser.parse(clara::Args(argc, argv));
+  if (!res) {
+    std::cerr << "Error in command line: " << res.errorMessage() << std::endl;
+    exit(1);
```

Comment:
If a user passes a device that does not exist you should give a warning saying that any device based on the order in your clinfo will be selected, you dont need to exit

Reply:
That's not what this code is doing, this code only checks that the --device parameter is valid (i.e. followed by an argument on the argument list).

In terms of legality checking, I think if the selector doesn't match exactly, the program should fail.

Reply:
The way device selectors are working, the program shouldn't fail. An invalid name will just result in a random device being used (best effort but with irrelevant search criteria)